<?xml version="1.0"?>

<launch>

    <arg name="scenario" default="sml" /> <!-- Empty scenario to make sure it is consistent if required -->
    <arg name="scenario_path" default="/home/sml/tiago_public_ws/src/rqt_simulation_gui/sml/" />

    <arg name="robot_name" default="robot1" />
    <arg name="initial_pose_x" default="0" />
    <arg name="initial_pose_y" default="0" />
    <arg name="initial_pose_a" default="0" />
    <arg name="robot_model" default="turtlebot" />

    <!-- The given behavior is automatically executed on start -->
    <arg name="behavior_name" default="Example Behavior" />

    <group ns="$(arg robot_name)">

        <!-- Run robot-specific software and spawn robot -->
        <include file="$(find rqt_simulation)/launch/robot_exp_$(arg robot_model).launch">
            <arg name="scenario" value="$(arg scenario)" />
            <arg name="scenario_path" value="$(arg scenario_path)" />
            <arg name="robot_name" value="$(arg robot_name)" />
            <arg name="initial_pose_x" value="$(arg initial_pose_x)" />
            <arg name="initial_pose_y" value="$(arg initial_pose_y)" />
            <arg name="initial_pose_a" value="$(arg initial_pose_a)" />
        </include>

    </group>

          <node name="rviz" pkg="rviz" type="rviz" args="-d $(find turtlebot_rviz_launchers)/rviz/navigation.rviz"/>

</launch>
